core/src/test/java/org/elasticsearch/index/query/TemplateQueryIT.java

```diff
@@ -187,16 +188,14 @@ public void testSearchRequestFail() throws Exception {
         searchRequest.indices("_all");
         try {
             String query = "{ \"template\" : { \"query\": {\"match_all\": {}}, \"size\" : \"{{my_size}}\"  } }";
-            BytesReference bytesRef = new BytesArray(query);
-            searchRequest.templateSource(bytesRef);
+            searchRequest.template(parseTemplate(query));
```

Comment:
I have to admit I don't feel we need to decide be that picky here it' really not relevant for this change at all and if somebody feels like these tests need to be cleaned up then they should be unittests instead. I think we can just stick with what we have and test all we can